File: nodes/dataset/load-dataset/load-dataset.js
module.exports = function(RED){
	function loadDatasetNode(config){
		const path = require('path')
		const utils = require('../../../utils/utils')

		var node = this;

		//set configurations
		node.file = __dirname + '/load-dataset.py'
		node.topic = 'sparkml'
		node.config = {
			// path: config.path,
			// input: Boolean(config.input),
			// output: Boolean(config.output)
		}

		utils.run(RED, node, config)
	}
	RED.nodes.registerType("load sparkml dataset", loadDatasetNode)
}

File: nodes/dataset/load-dataset/load-dataset.py
import json
import pyspark
import os, sys
from pyspark.sql import SparkSession
from pyspark.ml.tuning import TrainValidationSplit

#read configurations
config = json.loads(input())

while True:
	#wait request
	data = input()
	if data:
		# print("Loading Dataset")

		# print(" Spark Master : " + json.loads(data)['sparkMaster'])
		spark = SparkSession.builder.master(json.loads(data)['sparkMaster']).appName(json.loads(data)['sparkApp']).getOrCreate()


		# print(" Dataset Path : " + datasetPath)
		# Prepare training and test data.
		train, test = spark.read.parquet(json.loads(data)["currentTrain"]), spark.read.parquet(json.loads(data)["currentTest"])

		train.cache()
		test.cache()
		spark.stop()

		# print('Dataset Loaded')
		
		print(json.dumps(json.loads(data)))
	# else:
	# 	print('Nothing to load.', file=sys.stderr)